ioctl/test_ioctl.c

```diff
@@ -0,0 +1,394 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    Phoenix-RTOS standard library functions tests
+ *    HEADER:
+ *    - ioctl.h
+ *    TESTED:
+ *    - ioctl()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <phoenix/ioctl.h>
+#include <sys/msg.h>
+#include <signal.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <posix/utils.h>
+#include <unity_fixture.h>
+
+static const int32_t exp_flag_val = 0x12345678;
+static uint32_t port;
+static int file_desc, dev_desc;
+static volatile int thread_running = 1;
+
+typedef uint8_t test_ioctl_buf[64];
+
+#define MAX_FAIL       10
+#define PATH_TF        "ioctl_testFile"
+#define PATH_REG       "ioctl_testRegularFile"
+#define DEV_IOCTL_TEST "/dev/ioctlTest"
+/* T stands for test, do not confuse with tty */
+#define TEST_GRP ('T')
+
+#define TEST_IOCTL_SIG       _IOC(IOC_VOID, TEST_GRP, 0x01, 0)
+#define TEST_IOCTL_IN_VAL    _IOC(IOC_VOID, TEST_GRP, 0x02, sizeof(int32_t))
+#define TEST_IOCTL_IN        _IOC(IOC_IN, TEST_GRP, 0x03, sizeof(int32_t))
+#define TEST_IOCTL_IN_BIG    _IOC(IOC_IN, TEST_GRP, 0x04, sizeof(test_ioctl_buf))
+#define TEST_IOCTL_OUT       _IOC(IOC_OUT, TEST_GRP, 0x05, sizeof(int32_t))
+#define TEST_IOCTL_OUT_BIG   _IOC(IOC_OUT, TEST_GRP, 0x06, sizeof(test_ioctl_buf))
+#define TEST_IOCTL_INOUT     _IOC(IOC_INOUT, TEST_GRP, 0x07, sizeof(int32_t))
+#define TEST_IOCTL_INOUT_BIG _IOC(IOC_INOUT, TEST_GRP, 0x08, sizeof(test_ioctl_buf))
+
+
+static void *thread(void *arg)
```

Comment:
Let's prefix global variables/functions with `test_` in all our tests (shadowing is much less probable then) + use camelCase as it's suggested in our coding convention.
```suggestion
static const int32_t test_expFlagVal = 0x12345678;
static uint32_t test_port;
static int test_fileDesc, test_devDesc;
static volatile int test_threadRunning = 1;

typedef uint8_t test_ioctl_buf[64];

#define MAX_FAIL       10
#define PATH_TF        "ioctl_testFile"
#define PATH_REG       "ioctl_testRegularFile"
#define DEV_IOCTL_TEST "/dev/ioctlTest"
/* T stands for test, do not confuse with tty */
#define TEST_GRP ('T')

#define TEST_IOCTL_SIG       _IOC(IOC_VOID, TEST_GRP, 0x01, 0)
#define TEST_IOCTL_IN_VAL    _IOC(IOC_VOID, TEST_GRP, 0x02, sizeof(int32_t))
#define TEST_IOCTL_IN        _IOC(IOC_IN, TEST_GRP, 0x03, sizeof(int32_t))
#define TEST_IOCTL_IN_BIG    _IOC(IOC_IN, TEST_GRP, 0x04, sizeof(test_ioctl_buf))
#define TEST_IOCTL_OUT       _IOC(IOC_OUT, TEST_GRP, 0x05, sizeof(int32_t))
#define TEST_IOCTL_OUT_BIG   _IOC(IOC_OUT, TEST_GRP, 0x06, sizeof(test_ioctl_buf))
#define TEST_IOCTL_INOUT     _IOC(IOC_INOUT, TEST_GRP, 0x07, sizeof(int32_t))
#define TEST_IOCTL_INOUT_BIG _IOC(IOC_INOUT, TEST_GRP, 0x08, sizeof(test_ioctl_buf))


static void *test_thread(void *arg)
```